osu.Game/Screens/Play/SubmittingPlayer.cs

```diff
@@ -128,40 +128,49 @@ private bool handleTokenRetrieval()
 
             return true;
 
-            void handleTokenFailure(Exception exception)
+            void handleTokenFailure(Exception exception, bool displayNotification = false)
             {
                 tcs.SetResult(false);
 
-                if (HandleTokenRetrievalFailure(exception))
+                bool shouldExit = ShouldExitOnTokenRetrievalFailure(exception);
+
+                if (displayNotification || shouldExit)
                 {
+                    string whatWillHappen = shouldExit
+                        ? "You are not able to submit a score."
+                        : "The following score will not be submitted.";
+
                     if (string.IsNullOrEmpty(exception.Message))
-                        Logger.Error(exception, "Failed to retrieve a score submission token.");
+                        Logger.Error(exception, $"{whatWillHappen} Failed to retrieve a score submission token.");
```

Comment:
```diff
diff --git a/osu.Game/Screens/Play/SubmittingPlayer.cs b/osu.Game/Screens/Play/SubmittingPlayer.cs
index 0873f60791..ecb507f382 100644
--- a/osu.Game/Screens/Play/SubmittingPlayer.cs
+++ b/osu.Game/Screens/Play/SubmittingPlayer.cs
@@ -137,11 +137,11 @@ void handleTokenFailure(Exception exception, bool displayNotification = false)
                 if (displayNotification || shouldExit)
                 {
                     string whatWillHappen = shouldExit
-                        ? "You are not able to submit a score."
-                        : "The following score will not be submitted.";
+                        ? "Play in this state is not permitted."
+                        : "Your score will not be submitted.";
 
                     if (string.IsNullOrEmpty(exception.Message))
-                        Logger.Error(exception, $"{whatWillHappen} Failed to retrieve a score submission token.");
+                        Logger.Error(exception, $"Failed to retrieve a score submission token.\n\n{whatWillHappen}");
                     else
                     {
                         switch (exception.Message)
@@ -149,11 +149,11 @@ void handleTokenFailure(Exception exception, bool displayNotification = false)
                             case @"missing token header":
                             case @"invalid client hash":
                             case @"invalid verification hash":
-                                Logger.Log($"{whatWillHappen} Please ensure that you are using the latest version of the official game releases.", level: LogLevel.Important);
+                                Logger.Log($"Please ensure that you are using the latest version of the official game releases.\n\n{whatWillHappen}", level: LogLevel.Important);
                                 break;
 
                             case @"expired token":
-                                Logger.Log($"{whatWillHappen} Your system clock is set incorrectly. Please check your system time, date and timezone.", level: LogLevel.Important);
+                                Logger.Log($"Your system clock is set incorrectly. Please check your system time, date and timezone.\n\n{whatWillHappen}", level: LogLevel.Important);
                                 break;
 
                             default:

```

Something like this may read better, I think.

Ultimately designing a custom notification for this is probably warranted, at some point.

Reply:
Have applied verbatim.

> Ultimately designing a custom notification for this is probably warranted, at some point.

Not sure what this would entail.

Reply:
Mostly just adding an icon and what not.